samples/msal-node-samples/on-behalf-of/test/obo-aad.spec.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import puppeteer from "puppeteer";
+import { Screenshot, createFolder, setupCredentials } from "../../../e2eTestUtils/TestUtils";
+import { NodeCacheTestUtils } from "../../../e2eTestUtils/NodeCacheTestUtils";
+import { LabClient } from "../../../e2eTestUtils/LabClient";
+import { LabApiQueryParams } from "../../../e2eTestUtils/LabApiQueryParams";
+import { AppTypes, AzureEnvironments } from "../../../e2eTestUtils/Constants";
+import { enterCredentials, SCREENSHOT_BASE_FOLDER_NAME, validateCacheLocation, SAMPLE_HOME_URL } from "../../testUtils";
+import { ConfidentialClientApplication } from "@azure/msal-node";
+import * as dotenv from "dotenv";
+
+dotenv.config({
+    path: "../../../../.env",
+});
+
+const { acquireTokenByCode, acquireTokenObo } = require("../index");
+
+const webAppConfig = require("../config/WEB-APP.json");
+const webApiConfig = require("../config/WEB-API.json");
+
+if (process.env.OBO_WEB_APP_CLIENT_ID) {
+    webAppConfig.authOptions.clientId = process.env.OBO_WEB_APP_CLIENT_ID;
+}
+if (process.env.OBO_WEB_APP_CLIENT_SECRET) {
+    webAppConfig.authOptions.clientSecret = process.env.OBO_WEB_APP_CLIENT_SECRET;
+}
+
+if (process.env.OBO_WEB_API_CLIENT_ID) {
+    webApiConfig.authOptions.clientId = process.env.OBO_WEB_API_CLIENT_ID;
+    webApiConfig.webApiUrl = `api://${process.env.OBO_WEB_API_CLIENT_ID}/access_as_user`;
+}
+if (process.env.OBO_WEB_API_CLIENT_SECRET) {
+    webApiConfig.authOptions.clientSecret = process.env.OBO_WEB_API_CLIENT_SECRET;
+}
+if (process.env.OBO_WEB_API_TENANT_ID) {
+    webApiConfig.authOptions.authority = `https://login.microsoftonline.com/${process.env.OBO_WEB_API_TENANT_ID}`;
+    webApiConfig.discoveryKeysEndpoint = `https://login.microsoftonline.com/${process.env.OBO_WEB_API_TENANT_ID}/discovery/v2.0/keys`;
+}
```

Comment:
curios why we need these from the environment -can we not query the lab api to get retrieve them at test run time? (e.g.)

Reply:
I'm using the same web app and web api apps as the dotnet obo e2e tests... and they are not queryable in the lab. I'm not sure why. I've entered their app ids in the swagger api and nothing is returned.